src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -1409,7 +1417,7 @@ private int processHeadUpdate(SCMHeadEvent<?> event, TaskListener global, String
                     for (SCMSource source : p.getSCMSources()) {
                         if (event.isMatch(source)) {
                             for (SCMHead h : event.heads(source).keySet()) {
-                                if (p.getItem(h.getName()) == null) {
+                                if (p.getItemByBranchName(h.getName()) == null) {
```

Comment:
Not strictly necessary as getItem will pick up the old name, but removes the 1 in 33,554,432 chance if somebody for example has a branch called `feature/ux-1` and a branch called `feature-ux-1.0gorp7`